ambry-commons/src/main/java/com.github.ambry.commons/ByteBufferAsyncWritableChannel.java

```diff
@@ -31,10 +31,46 @@
  * retrieved and resolved by an external thread.
  */
 public class ByteBufferAsyncWritableChannel implements AsyncWritableChannel {
+
+  /**
+   * List of events of interest to the consumer of the content in this channel.
+   */
+  public enum EventType {
+    ChunkArrival,
+    Close,
+  }
+
+  /**
+   * Callback that can be used to listen to events that happen inside this channel.
+   */
+  public interface ChannelEventListener {
```

Comment:
Just would like to confirm that even making `ChannelEventListener` and `EventType` public, they are still within `ByteBufferAsyncWritableChannel` because of the logic relationship? I.e., the interface and enum are only meaningful to `ByteBufferAsyncWritableChannel`?